File: Examples/VisualBasic/RunExamples.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.IO
Imports Aspose.OCR.Examples.VisualBasic.FormattingAndManipulatingOCR
Imports Aspose.OCR.Examples.VisualBasic.FormattingAndManipulatingOMR
Imports Aspose.OCR.Examples.VisualBasic.PerformingandManagingOCR

Namespace Aspose.OCR.Examples.VisualBasic
    Class RunExamples
        Public Shared Sub Main(args As String())
            Console.WriteLine("Open RunExamples.vb. " & vbLf & "In Main() method Uncomment the example that you want to run.")
            Console.WriteLine("=====================================================")

            ' Uncomment the example that you want to run.

            ' =====================================================
            ' =====================================================
            '  Performing And Managing OCR
            ' =====================================================
            ' =====================================================

            'PerformOCROnImage.Run()
            'GettingNotification.Run()
            'ExtractingText.Run()
            'GetLocationAndSizeExample.Run()
            'ReadPartInformation.Run()
            'DetectingTextBlocks.Run()
            'GetTextPartHierarchy.Run()
            'LoadTemplateFromUrl.Run()
            'WorkingWithDifferentLanguages.Run()
            'OCROnMultipageTIFF.Run()
            'PreprocesImagesFromOCROperation.Run()
            'SupplyMultipleWhiteLists.Run()

            ' =====================================================
            ' =====================================================
            ' Formatting And Manipulating OCR   
            ' =====================================================
            ' =====================================================

            'AddingUserDefinedRecognitionBlocks.Run()
            'ApplyingCorrectionFilters.Run()
            'AutomaticallyCorrectTheSpellings.Run()
            'PreprocesImagesFromOCROperation.Run()
            'RecognizeWhiteListedCharacters.Run()
            'SettingOcrEngineAutomaticallyDetect.Run()
            'SettingtheOcrEngine.Run()
            'SettingtheOcrEngineAutomaticallyDetectTextBlocks.Run()
            'UserDefinedRecognitionBlocks.Run()
            'ExtractPreprocessedImages.Run()
            'UsingNotifierFactory.Run()

            ' =====================================================
            ' =====================================================
            ' Formatting And Manipulating OMR   
            ' =====================================================
            ' =====================================================

            'DetectImageResolutionAutomatically.Run()
            'DetectUsingElements.Run()
            'ExtractOMRDataUsingScannedImage.Run()
            'ExtractText.Run()
            'PerformOCROnImage.Run()
            'SetElementLevelThreshold.Run()
            'SetImageResolution.Run()
            'SetPageLevelThreshold.Run()
            'SettingMarkThreshold.Run()
            'SkewImageCorrectionUsingAlgorithm.Run()
            'SkewedImageSecondMethod.Run()
            'DynamicallyCreateOMRTemplates.Run()
            'AdjustImageProcessPipeline.Run()

            ' Stop before exiting
            Console.WriteLine(Environment.NewLine + "Program Finished. Press any key to exit....")
            Console.ReadKey()
        End Sub


        Public Shared Function GetDataDir_OMR() As [String]
            Return Path.GetFullPath(GetDataDir_Data() & Convert.ToString("OMR/"))
        End Function

        Public Shared Function GetDataDir_OCR() As [String]
            Return Path.GetFullPath(GetDataDir_Data() & Convert.ToString("OCR/"))
        End Function

        Private Shared Function GetDataDir_Data() As String
            Dim parent = Directory.GetParent(Directory.GetCurrentDirectory()).Parent
            Dim startDirectory As String = Nothing
            If parent IsNot Nothing Then
                Dim directoryInfo = parent.Parent
                If directoryInfo IsNot Nothing Then
                    startDirectory = directoryInfo.FullName
                End If
            Else
                startDirectory = parent.FullName
            End If
            Return If(startDirectory IsNot Nothing, Path.Combine(startDirectory, "Data\"), Nothing)
        End Function

    End Class
End Namespace

File: Examples/CSharp/RunExamples.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;
using Aspose.OCR.Examples.CSharp.PerformingandManagingOCR;

namespace Aspose.OCR.Examples.CSharp
{
    class RunExamples
    {
        static void Main(string[] args)
        {
            //License ocrLicense = new License();
            //ocrLicense.SetLicense("LicenseFilePath");


            Console.WriteLine("Open RunExamples.cs. \nIn Main() method Uncomment the example that you want to run.");
            Console.WriteLine("=====================================================");

            // Uncomment the example that you want to run.

            //// =====================================================
            //// =====================================================
            ////  Performing And Managing OCR
            //// =====================================================
            //// =====================================================

            //PerformOCROnImage.Run();
            //RecognizeLine.Run();
            //WorkingWithDifferentLanguages.Run();
            //RecognizeImageFromStream.Run();
            //RecognizeImageWithoutTextAreaDetection.Run();
            //RecognizeImageWithTextAreaDetectionAndWithoutSkewCorrection.Run();
            //PrepareRectangles.Run();
            //RecognizeTextFromSpecificRectangle.Run();
            //SpecifyAllowedCharacters.Run();
            //CalculateSkewAngle.Run();
            //CalculateSkewAngleFromStream.Run();

            // Stop before exiting
            Console.WriteLine(Environment.NewLine + "Program Finished. Press any key to exit....");
            Console.ReadKey();
        }

        public static String GetDataDir_OCR()
        {
            return Path.GetFullPath(GetDataDir_Data() + "OCR/");
        }

        private static string GetDataDir_Data()
        {
            var parent = Directory.GetParent(Directory.GetCurrentDirectory()).Parent;
            string startDirectory = null;
            if (parent != null)
            {
                var directoryInfo = parent.Parent;
                if (directoryInfo != null)
                {
                    startDirectory = directoryInfo.FullName;
                }
            }
            else
            {
                startDirectory = parent.FullName;
            }
            return startDirectory != null ? Path.Combine(startDirectory, "Data\\") : null;
        }

    }
}